writers/writer.go

```diff
@@ -57,22 +66,82 @@ func (c *consumer) consume(m *nats.Msg) {
 	}
 	var msgs []senml.Message
 	for _, v := range norm {
-		if c.channelExists(v.Channel) {
+		if c.channelExists(v.Channel) && c.subtopicExists(v.Subtopic) {
 			msgs = append(msgs, v)
 		}
 	}
 
+	if msgs == nil {
+		return
+	}
+
 	if err := c.repo.Save(msgs...); err != nil {
 		c.logger.Warn(fmt.Sprintf("Failed to save message: %s", err))
 		return
 	}
 }
 
 func (c *consumer) channelExists(channel string) bool {
-	if _, ok := c.channels["*"]; ok {
-		return true
+	for _, ch := range c.channels {
+		if ch == channel || ch == "*" {
+			return true
+		}
 	}
+	return false
+}
+
+func (c *consumer) subtopicExists(subtopic string) bool {
+	for _, s := range c.subtopics {
+		if s == subtopic || s == "*" {
+			return true
+		}
+	}
+	return false
+}
+
+type filterConfig struct {
+	List []string `toml:"filter"`
+}
+
+type channelsConfig struct {
+	Channels filterConfig `toml:"channels"`
+}
+
+type subtopicsConfig struct {
+	Subtopics filterConfig `toml:"subtopics"`
+}
+
+type FiltersCfg struct {
+	channels  []string
+	subtopics []string
+}
+
+func LoadFiltersConfig(channelConfigPath string, subtopicConfigPath string) (FiltersCfg, error)  {
+	filters := FiltersCfg{}
+
+	data, err := ioutil.ReadFile(channelConfigPath)
+	if err != nil {
+		return filters, errors.Wrap(errOpenConfFile, err)
+	}
+
+	var channelsCfg channelsConfig
+	if err := toml.Unmarshal(data, &channelsCfg); err != nil {
+		return filters, errors.Wrap(errParseConfFile, err)
+	}
+
+	filters.channels = channelsCfg.Channels.List
+
+	data, err = ioutil.ReadFile(subtopicConfigPath)
+	if err != nil {
+		return filters, errors.Wrap(errOpenConfFile, err)
+	}
+
+	var subtopicCfg subtopicsConfig
+	if err := toml.Unmarshal(data, &subtopicCfg); err != nil {
+		return filters, errors.Wrap(errParseConfFile, err)
+	}
+
+	filters.subtopics = subtopicCfg.Subtopics.List
 
-	_, found := c.channels[channel]
-	return found
+	return filters, err
```

Comment:
Sorry again, but this can be simplified with:
```
return FiltersCfg{
    channels: channelsCfg.Channels.List,
    subtopics: subtopicsCfg.Subtopics.List,
  }, err
```

If you have errors you can do: `return FiltersCfg{}, errors.Wrap(errOpenConfFile, err)`